### PR TITLE
fix: Handle unverified account error

### LINKED ISSUE #2286: Handle error for unverified email
**Describe the bug**
Redirect to profile if an unverified user tries to buy a free ticket. 

**Would you like to work on the issue?**

yes

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #2286 
<p align="center">
 
</p>

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt [hunk 1]
```diff
@@ -165,6 +167,11 @@ class AttendeeViewModel(
                 mutablePendingOrder.value = it
                 orderIdentifier = it.identifier.toString()
             }, {
+                if (it is HttpException) {
+                    if (it.code() == HttpErrors.UNPROCESSABLE_ENTITY) {
+                        mutableRedirectToProfile.value = true
```
Comment (ShridharGoel) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: But this error can be due to other reasons too.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Then what should be the solution?

Comment (ShridharGoel) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Check the pointer field of the error and if it indicates unverified account error, then redirect the user to profile.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: {status: 422, source: {pointer: "/data/relationships/order"}, title: "Unprocessable Entity",…}

Comment (ShridharGoel) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: And it does not have any message?

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: I'm not sure if it is okay to compare the message: detail: "Unverified user cannot place free orders"

Comment (ShridharGoel) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: @iamareebjamal I think there should have been more information in the pointer.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Check if message containsIgnoreCase 'unverified user' for now. Open an issue on server to change the error to `Unauthorized` from `Unprocessable Entity` and correct the pointer. Both are wrong

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Updated

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt [hunk 2]
```diff
@@ -165,6 +168,11 @@ class AttendeeViewModel(
                 mutablePendingOrder.value = it
                 orderIdentifier = it.identifier.toString()
             }, {
+                if (it is HttpException) {
+                    if (ErrorUtils.getErrorDetails(it).detail?.contains(UNVERIFIED_USER, true) == true) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: containsIgnoreCase?

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: > containsIgnoreCase?

Ignore case is set to true. `contains(UNVERIFIED_USER, true)`

**Pull request merged**